R/methods-pSet.R

```diff
@@ -546,10 +546,29 @@ setReplaceMethod("$", "pSet", function(x, name, value) {
 })
 
 setReplaceMethod("pData", "pSet", function(object, value) {
+    if (!is.data.frame(value))
+        stop("'value' has to be a 'data.frame'")
     pData(object@phenoData) <- value
     object
 })
 
+setReplaceMethod("phenoData", "pSet", function(object, value) {
+    ## Check input and convert to NAnnotatedDataFrame if neccessary
+    if (is.data.frame(value)) {
```

Comment:
I think it would be better that `phenoData<-` worked with an `AnnotatedDataFrame` and a `NAnnotatedDataFrame`, and that `pData<-` was reserved for a `data.frame`. I think this is closer to the semantic of the accessorts.

Reply:
OK